Objects/dictobject.c

```diff
@@ -7054,6 +7093,79 @@ _PyObject_TryGetInstanceAttribute(PyObject *obj, PyObject *name, PyObject **attr
 #endif
 }
 
+bool
+_PyObject_TryGetInstanceAttributeStackRef(PyObject *obj, PyObject *name, _PyStackRef *attr)
+{
+    assert(PyUnicode_CheckExact(name));
+    PyDictValues *values = _PyObject_InlineValues(obj);
+    if (!FT_ATOMIC_LOAD_UINT8(values->valid)) {
+        return false;
+    }
+
+    PyDictKeysObject *keys = CACHED_KEYS(Py_TYPE(obj));
+    assert(keys != NULL);
+    Py_ssize_t ix = _PyDictKeys_StringLookup(keys, name);
+    if (ix == DKIX_EMPTY) {
+        *attr = PyStackRef_NULL;
+        return true;
+    }
+
+#ifdef Py_GIL_DISABLED
+    PyObject *value = _Py_atomic_load_ptr_acquire(&values->values[ix]);
+    *attr = PyStackRef_FromPyObjectNew(value);
```

Comment:
`PyStackRef_FromPyObjectNew(value)` is not thread-safe here

Reply:
We need the TryXGetRef pattern right?

Reply:
Yes, but with `_PyStackRef`. I think something like:

```c
static inline _PyStackRef
_Py_TryXGetStackRef(PyObject **ptr)
{
    PyObject *value = _Py_atomic_load_ptr(ptr);
    if (value == NULL) {
        return PyStackRef_NULL;
    }
    if (_Py_IsImmortal(value) || _PyObject_HasDeferredRefcount(value)) {
        return (_PyStackRef){ .bits = (uintptr_t)value | Py_TAG_DEFERRED };
    }
    if (_Py_TryIncrefCompare(ptr, value)) {
        return PyStackRef_FromPyObjectSteal(value);
    }
    return PyStackRef_NULL;
}
```